src/vcpkg/base/system.process.cpp

```diff
@@ -246,6 +249,40 @@ namespace vcpkg
 #endif
     }
 
+    void get_parent_process_list(std::vector<std::string>& ret)
+    {
+#if defined(_WIN32)
+        struct ProcessInformation
+        {
+            NTSTATUS exit_status;
+            PPEB peb_base_address;
+            ULONG_PTR affinity_mask;
+            KPRIORITY base_priority;
+            ULONG_PTR unique_process_id;
+            ULONG_PTR inherited_from_unique_process_id;
+        } process_info;
+
+        wchar_t process_name_buf[_MAX_PATH];
+        HANDLE process_handle = GetCurrentProcess();
+        while (true)
+        {
+            auto has_error = NtQueryInformationProcess(
```

Comment:
Are we OK for apiscan purposes touching this? That we needed a struct that isn't in the Windows SDK is a bit 😬 on that front...

Reply:
The structure itself is documented here:
* [NtQueryInformationProcess | PROCESS_BASIC_INFORMATION](https://learn.microsoft.com/en-us/windows/win32/api/winternl/nf-winternl-ntqueryinformationprocess#process_basic_information)

But the whole article has a disclaimer about using an internal API.

Reply:
@BillyONeal 

7d5e04878228949d5a0ff9b061fe62b7f4ba4132 uses a different API, the trade-off is that we have to enumerate all processes and then look for the vcpkg parent process hierarchy in a second step.

Reply:
> The structure itself is documented here:
> 
> * [NtQueryInformationProcess | PROCESS_BASIC_INFORMATION](https://learn.microsoft.com/en-us/windows/win32/api/winternl/nf-winternl-ntqueryinformationprocess#process_basic_information)

OK, that sounds like documented to me. 👍